gf-webapp/src/component/filter/RoomsBlockWithFilters.tsx

```diff
@@ -0,0 +1,99 @@
+import {useEffect, useState} from "react";
+import {Room} from "src/model/Room/Room";
+import {RoomItem} from "src/logic/rooms/roomsPage/roomBlock/roomItem/RoomItem";
+import {RoomApiService} from "src/service/RoomApi/RoomApi";
+import {useFilterRooms} from "src/component/filter/FilterContext";
+
+export const RoomsBlockWithFilters = () => {
+  const {searchValue, adultsValue, childrenValue, dateInValue, dateOutValue} = useFilterRooms();
+
+  const [rooms, setRooms] = useState<Room[] | null>(null);
+
+  const initRooms = async () => {
+    const roomsList = await RoomApiService.getAllRooms();
+    setRooms(roomsList);
+  };
+
+  useEffect(() => {
+    async function onRoomsInitialized() {
+      await initRooms();
+    }
+    onRoomsInitialized();
+  }, []);
+
+  const adultsFilterCondition = (roomFiltering: Room) => {
+    if (adultsValue !== 0) {
+      return roomFiltering.adults === adultsValue;
+    } else {
+      return roomFiltering;
+    }
+  };
+
+  const childrenFilterCondition = (roomFiltering: Room) => {
+    if (childrenValue !== 0) {
+      return roomFiltering.children === childrenValue;
+    } else {
+      return roomFiltering;
+    }
+  };
+
+  const searchFilterCondition = (roomFiltering: Room) => {
+    return Object.values(roomFiltering)
+      .join()
+      .toString()
+      .toLowerCase()
+      .includes(searchValue);
+  };
+
+  const dateFilterCondition = (roomFiltering: Room) => {
+    if (new Date(dateInValue).getTime() >= new Date(dateOutValue).getTime()
+      || new Date(dateInValue).getTime() < new Date().getTime()
+      || new Date(dateOutValue).getTime() < new Date().getTime()) {
+      alert("Wrong dates!!!");
```

Comment:
Nice

Reply:
Create separate function to getTime from any date